tests/virtual_machines_test.go

```diff
@@ -516,26 +516,54 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 		Context("When the leader is changed", func() {
 			It("should be able to create a new virtual machine", func() {
+				By("Restart kubemacpool to reset cache and mac range")
 				err := initKubemacpoolParams(rangeStart, rangeEnd)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).ToNot(HaveOccurred(), "should success restarting kubemacpool to reset mac range and cache")
 
 				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "")},
 					[]kubevirtv1.Network{newNetwork("br")})
 
 				anotherVm := vm.DeepCopy()
 				anotherVm.Name = "another-vm"
 
+				By("Create VM")
 				err = testClient.VirtClient.Create(context.TODO(), vm)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).ToNot(HaveOccurred(), "should success creating the vm")
 				_, err = net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).ToNot(HaveOccurred(), "should success parsing the vm's mac")
 
 				ChangeManagerLeadership()
 
+				By("checking that a new VM can be created on new leader")
 				err = testClient.VirtClient.Create(context.TODO(), anotherVm)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).ToNot(HaveOccurred(), "should success creating the vm")
 				_, err = net.ParseMAC(anotherVm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).ToNot(HaveOccurred(), "should success parsing the vm's mac")
+			})
+			It("should successfully prevent collisions with virtual machine created on previous leader", func() {
+				By("Restart kubemacpool to reset cache and mac range")
+				err := initKubemacpoolParams(rangeStart, rangeEnd)
+				Expect(err).ToNot(HaveOccurred(), "should success restarting kubemacpool to reset mac range and cache")
+
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "02:00:ff:ff:ff:ff")},
+					[]kubevirtv1.Network{newNetwork("br")})
+
+				anotherVm := vm.DeepCopy()
+				anotherVm.Name = "another-vm"
+
+				By("Create VM")
+				err = testClient.VirtClient.Create(context.TODO(), vm)
+				Expect(err).ToNot(HaveOccurred(), "should success creating the vm")
+				By("checking that the new vm's allocated mac is valid")
+				_, err = net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred(), "should success parsing the vm's mac")
+
+				ChangeManagerLeadership()
+
+				By("checking that a new VM with same mac as old vm cannot be created")
+				err = testClient.VirtClient.Create(context.TODO(), anotherVm)
+				Expect(err).To(HaveOccurred(), "should fail creating vm with duplicate mac")
+				Expect(strings.Contains(err.Error(), "failed to allocate requested mac address")).To(Equal(true))
```

Comment:
We should avoid checks like these please change/remove it.
Depending on specific error string is bad practice, we had  a test that checked for specific error string, that error string eventually changed and that test became false positive.
The time that it took to debug and and fix it could have been spent on something more important :wink: